posts/distributed-tracing-with-xray.md

```diff
@@ -0,0 +1,269 @@
+                "protocol": "udp"
+              }
+            ]
+          },
+          {
+            "name": "micro1-container",
+            "image": "9666461757999.dkr.ecr.us-east-2.amazonaws.com/micro1:65435390a2d95f73f6f379cb63338a305641987ba",
+            "cpu": 512,
+            "portMappings": [
+              {
+                "hostPort": 8888,
+                "protocol": "tcp",
+                "containerPort": 8888
+              }
+            ],
+            "memory": 1024,
+            "essential": true,
+            "logConfiguration": {
+              "logDriver": "awslogs",
+              "options": {
+                "awslogs-group": "/ecs/group1",
+                "awslogs-region": "us-east-2",
+                "awslogs-stream-prefix": "ecs"
+              }
+            }
+          }
+        ],
+        "requiresCompatibilities": [
+          "FARGATE"
+        ],
+        "networkMode": "awsvpc",
+        "cpu": "1024",
+        "memory": "2048"
+      }
+
+    ```
+    
+Please checkout this [blog](https://blog.ippon.tech/deploytoecswithgithubaction/) on how to deploy a spring boot microservice on Amazon ECS using Github Actions
+
+
+# View and Analyse the tracing data on AWS X-Ray Console
+X-Ray console uses trace data sent by the X-Ray daemon to generate a detailed service graph. The service graph can shows the client, front-end service, and backend services.
+
+In this example we are tracing the client requests as they travel through the application which is composed by 2 microservices
+
+![02](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/02/X-Ray-console.png)
+
+The console provides a filter expression language for filtering requests, and services based on data in request headers, response status, and indexed fields on the tracing data.
+
+The service graph helps identify bottlenecks, latency spikes as shown in the example bellow where we introduced a latency on the first microservice.
+
+![03](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/02/X-Ray-console2.png)
+
+It also identify services where errors are occurring
+
+![04](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/02/X-Ray-console3.png)
+
+Exception logs are also available
+
+![05](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2021/02/X-Ray-console4.png)
+
+More details on the AWS X-Ray console can be found [here](https://docs.aws.amazon.com/xray/latest/devguide/xray-console.html)
+
+# Conclusion
+AWS X-Ray integrate natively with other AWS Services to increase the visibility on your Architecture, the [AWS X-Ray agent can also be run locally](https://docs.aws.amazon.com/xray/latest/devguide/xray-daemon-local.html) if some pieces of the application is deployed on-premises.
+
+AWS X-Ray always encrypts traces at rest, and can also be configured to use AWS Key Management Service (AWS KMS) for compliance requirement like PCI and HIPAA.
```

Comment:
Do we need a call-to-action here?

Reply:
Can you check with Youssra on what we should put here?

Reply:
Done